kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java

```diff
@@ -0,0 +1,48 @@
+package com.kodilla.patterns.prototype.library;
+
+import com.kodilla.patterns.prototype.clone.Prototype;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Library extends Prototype {
+    private String name;
+    private Set<Book> books = new HashSet<>();
+
+    public Library(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Set<Book> getBooks() {
+        return books;
+    }
+
+    public Library shallowCopy() throws CloneNotSupportedException {
+        return (Library) super.clone();
+    }
+
+    public Library deepCopy() throws CloneNotSupportedException {
+        Library clonedLibrary = shallowCopy();
+        clonedLibrary.books = new HashSet<>();
+
+        books.forEach(i -> clonedLibrary.getBooks().add(i));
```

Comment:
łamiesz tutaj tzw. prawo Demeter :) 
Dodaj metodę addBook(Book book) w klasie Library.
Tutaj nie ma deep copy! Jeśli zmienię tytuł książki w library, to zmieni się też tytuł tej książki w clonedLibrary.